### PR TITLE
Chore: Refactor useLicenseLimits() hook

### PR DESCRIPTION
**Pull request by gu-stav**
### What does it do?

Simplifies the `useLicenseLimits()` hook

### Why is it needed?

- by returning an object always we can simplify the destructuring
- by flattening the `license` object the consuming code becomes easier to read
- by returning `isLoading` components can handle that state
- by returning `isError` components can handle that state

### CODE REVIEW

File: docs/docs/docs/01-core/admin/01-ee/02-hooks/use-license-limits.mdx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+## Usage
+
+```js
+const { license, isError, isLoading } = useLicenseLimits();
+```
+
+### `license`
+
+An object the contains the whole admin API response.
```
Comment (joshuaellis) [inline docs/docs/docs/01-core/admin/01-ee/02-hooks/use-license-limits.mdx:21]: Can we type the object? even if it's a first draft / rough example?

**Review by joshuaellis — DISMISSED** [review]

**Review by jhoward1994 — DISMISSED** [review]

File: docs/docs/docs/01-core/admin/01-ee/02-hooks/use-license-limits.mdx (hunk 1 above)
Comment (gu-stav) [reply docs/docs/docs/01-core/admin/01-ee/02-hooks/use-license-limits.mdx:21]: Done & rebased.

**Review by joshuaellis — DISMISSED** [review]

**Review by joshuaellis — APPROVED** [review]

**Review by joshuaellis — APPROVED** [review]

**Review by joshuaellis — COMMENTED** [review]
Shall we squash this before merge?

File: packages/core/admin/ee/admin/hooks/useLicenseLimits/useLicenseLimits.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    }
+  );
+
+  const license = data ?? {};
+
+  const getFeature = (name) => {
+    const feature = (license?.features ?? []).find((feature) => feature.name === name);
+
+    return feature?.options ?? {};
+  };
```
[the remark this answers is not in the document]
Comment (gu-stav) [reply packages/core/admin/ee/admin/hooks/useLicenseLimits/useLicenseLimits.js]: Done & rebased.

**Comment by gu-stav** [comment]
> Shall we squash this before merge?

I think it is not necessary - each commit has it's own value and scope. I've merged 3 commits with fixes.

File: packages/core/admin/ee/admin/hooks/useLicenseLimits/useLicenseLimits.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+import { selectAdminPermissions } from '../../../../admin/src/pages/App/selectors';
+
+export function useLicenseLimits() {
+  const permissions = useSelector(selectAdminPermissions);
+  const { get } = useFetchClient();
+  const {
+    isLoading: isRBACLoading,
+    allowedActions: { canRead, canCreate, canUpdate, canDelete },
+  } = useRBAC(permissions.settings.users);
```
Comment (joshuaellis) [inline packages/core/admin/ee/admin/hooks/useLicenseLimits/useLicenseLimits.js:15]: in the case of RW workflows, would this mean the hook would not run if you couldn't update a user?

Comment (gu-stav) [reply packages/core/admin/ee/admin/hooks/useLicenseLimits/useLicenseLimits.js:15]: Seems so. I didn't invent this.

Comment (joshuaellis) [reply packages/core/admin/ee/admin/hooks/useLicenseLimits/useLicenseLimits.js:15]: No previously this hook was only used for admin seats.... i think in RW we'll need to tweak it so you can pass `enabled` as a prop 🤔

**Review by joshuaellis — APPROVED** [review]

File: packages/core/admin/ee/admin/hooks/useLicenseLimits/useLicenseLimits.js (hunk 3 above)
Comment (gu-stav) [reply packages/core/admin/ee/admin/hooks/useLicenseLimits/useLicenseLimits.js:15]: Let's do some pair programming on this, when we start implementing RBAC! 👍🏼

**Pull request merged**